python_modules/dagster/dagster_tests/components_tests/executable_component_tests/test_executable_component_in_memory.py

```diff
@@ -29,8 +32,7 @@ def _execute_fn(context) -> MaterializeResult:
         return MaterializeResult(metadata={"foo": "bar"})
 
     component = ExecutableComponent(
-        name="op_name",
-        execute_fn=_execute_fn,
+        execution=ExecutionModel(name="op_name", fn=_execute_fn),
```

Comment:
"Model" feels weird as user facing suffix to me. `Execution` or `ExecutionSpec` maybe? 

`execution=Execution(name="op_name", fn=_execute_fn)`
`execution=Execution(_execute_fn, name="op_name")` (allow only fn as pos args)
`execution=_execute_fn` (union support mentioned above)

Reply:
The component being called "ExecutableComponent" is mixed up in this as well. Part of me enjoys using the same root word in different ways and part of me thinks its confusing. 

The "resolved" system is a `Resolvable` base class with `Resolver` annotated fields 

`ExecutableComponent`  defines an `Execution` and the metadata about the assets/checks it computes 

is `Compute` any good? Feel like there was solid internals called compute for a long time

Reply:
were going all the way back https://github.com/dagster-io/dagster/pull/77 `ComputeNodeComponent`